socialhome/users/tests/test_views.py

```diff
@@ -111,32 +110,49 @@ def test_get_json_context(self):
                 "profile": {
                     "id": profile.id,
                     "guid": profile.guid,
-                    "followersCount": 3,
-                    "followingCount": str(profile.following.count()),
+                    "followers_count": 0,
+                    "following_count": profile.following.count(),
                     "handle": profile.handle,
-                    "saferImageUrlLarge": profile.safer_image_url_large,
-                    "streamType": view.profile_stream_type,
-                    "pinnedContentExists": view.pinned_content_exists,
+                    "has_pinned_content": Content.objects.profile_pinned(profile, request.user).exists(),
+                    "home_url": profile.home_url,
+                    "image_url_large": profile.image_url_large,
+                    "image_url_medium": profile.image_url_medium,
+                    "image_url_small": profile.image_url_small,
+                    "is_local": profile.is_local,
+                    "location": profile.location,
+                    "name": profile.name,
+                    "nsfw": profile.nsfw,
+                    "stream_type": view.profile_stream_type,
+                    "url": profile.url,
+                    "visibility": str(profile.visibility).lower(),
                 },
             },
         )
         request, view, contents, profile = self._get_request_view_and_content(anonymous_user=True, create_content=False)
-        view.followers_count = 3
         self.assertEqual(
             view.get_json_context(),
             {
-                "currentBrowsingProfileId": None,
+                "currentBrowsingProfileId": request.user.id,
```

Comment:
Test is failing. Maybe because you're using a user ID, not profile ID?

Reply:
Shouldn't be the same?

Reply:
No, local users have a `User`. Remote users don't. But everybody has a `Profile`. So what you should be setting (assuming a user is logged in) is `request.user.profile_id`. If it's an anonymous user, then pass in `None`, since anonymous users don't have a profile.

Reply:
This doesn't seem right. If the test passes it's probably because the ID's happen to match, not because they will always match. `currentBrowsingProfileId` should reference a `Profile` object ID. Does the test pass if you set it as `request.user.profile_id`?

Reply:
Ah, this is why:

```
In [4]: AnonymousUser().id is None
Out[4]: True
```

Still, this would be better more explicit to say `"currentBrowsingProfileId": None,` as it was before.

Reply:
`AnonymousUser` has no attribute `profile_id`. If I put `None` here, the test passes. I suppose `request.user.id` is `None` when using `AnonymouseUser`.

Reply:
Yeah the test should just check for the values it expects so just compare to `None` 👍